packages/ses/src/compartment.js

```diff
@@ -172,18 +175,63 @@ defineProperties(InertCompartment, {
  * @returns {Compartment['constructor']}
  */
 
+// In order to facilitate migration from the deprecated signature
+// of the compartment constructor,
+//   new Compartent(globals?, modules?, options?)
+// to the new signature:
+//   new Compartment(options?)
+// where globals and modules are expressed in the options bag instead of
+// positional arguments, this function detects the temporary sigil __options__
+// on the first argument and coerces compartments arguments into a single
+// compartments object.
+const compartmentOptions = (...args) => {
+  if (args.length === 0) {
+    return {};
+  }
+  if (
+    args.length === 1 &&
+    typeof args[0] === 'object' &&
+    args[0] !== null &&
+    '__options__' in args[0]
+  ) {
+    const { __options__, ...options } = args[0];
+    assert(
+      __options__ === true,
+      `Compartment constructor only supports true __options__ sigil, got ${__options__}`,
+    );
+    return options;
+  } else {
+    const [
+      globals = /** @type {Map<string, any>} */ ({}),
+      modules = /** @type {Map<string, ModuleDescriptor>} */ ({}),
+      options = {},
+    ] = args;
+    assertTypeof(
+      options.modules,
+      'undefined',
```

Comment:
Are you sure you want to check that the value is `undefined` rather than whether the property is present?

Why typeof? The above code should be equivalent to
```suggestion
    assertEqual(
      options.modules,
      undefined,
```
but the later says what it means better.